### PR TITLE
Add trace and span_id to logging async API

### PR DESCRIPTION
**Pull request by brunoais**
Follow up to #5885 and #5878. Add function parameters on the logging async API to allow API logging with trace and span_id in a non-blocking way.

### CODE REVIEW

**Comment by brunoais** [comment]
Do I need to add any tests for these changes? If so, may someone please guide me to where such files are?
In order to use stackdriver effectively, I really need this feature.

**Comment by brunoais** [comment]
>            for item in items:
 >               if item is _WORKER_TERMINATOR:
>                    quit_ = True
>                    # Continue processing items, don't break, try to process
>                    # all items we got back before quitting.
>                else:
>                   batch.log_struct(**item)
> E                   TypeError: log_struct() got an unexpected keyword argument 'trace'

but `trace` is there (same for `span_id`) :
https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/logging/google/cloud/logging/logger.py#L475

Am I seeing things wrong?

File: logging/google/cloud/logging/handlers/transports/background_thread.py [hunk 1]
```diff
@@ -229,7 +229,7 @@ def _main_thread_terminated(self):
                 'Failed to send %d pending logs.' % (self._queue.qsize(),),
                 file=sys.stderr)
 
-    def enqueue(self, record, message, resource=None, labels=None):
+    def enqueue(self, record, message, resource=None, labels=None, trace=None, span_id=None):
```
Comment (tseaver) [inline logging/google/cloud/logging/handlers/transports/background_thread.py]: Wrap this line (79 characters max).

File: logging/google/cloud/logging/handlers/transports/background_thread.py [hunk 2]
```diff
@@ -296,7 +305,7 @@ def __init__(self, client, name, grace_period=_DEFAULT_GRACE_PERIOD,
                               max_latency=max_latency)
         self.worker.start()
 
-    def send(self, record, message, resource=None, labels=None):
+    def send(self, record, message, resource=None, labels=None, trace=None, span_id=None):
```
Comment (tseaver) [inline logging/google/cloud/logging/handlers/transports/background_thread.py]: Wrap this line (79 characters max).

File: logging/google/cloud/logging/handlers/transports/background_thread.py [hunk 3]
```diff
@@ -311,8 +320,15 @@ def send(self, record, message, resource=None, labels=None):
...
+
+        :type trace: str
+        :param trace: (optional) traceid to apply to the logging entry.
+
+        :type span_id: str
+        :param span_id: (optional) span_id within the trace for the log entry.
+                        Specify the trace parameter if span_id is set.
         """
-        self.worker.enqueue(record, message, resource=resource, labels=labels)
+        self.worker.enqueue(record, message, resource=resource, labels=labels, trace=trace, span_id=span_id)
```
Comment (tseaver) [inline logging/google/cloud/logging/handlers/transports/background_thread.py]: Wrap this line (79 characters max).

File: logging/google/cloud/logging/handlers/transports/background_thread.py (hunk 1 above)
Comment (tseaver) [inline logging/google/cloud/logging/handlers/transports/background_thread.py]: Wrap this line (79 characters max).

File: logging/google/cloud/logging/handlers/transports/background_thread.py (hunk 2 above)
Comment (tseaver) [inline logging/google/cloud/logging/handlers/transports/background_thread.py]: Wrap this line (79 characters max).

File: logging/google/cloud/logging/handlers/transports/background_thread.py (hunk 3 above)
Comment (tseaver) [inline logging/google/cloud/logging/handlers/transports/background_thread.py]: Wrap this line (79 characters max).

File: logging/tests/unit/handlers/transports/test_background_thread.py [hunk 4]
```diff
@@ -64,7 +64,49 @@ def test_send(self):
...
+
+        transport, _ = self._make_one(client, name)
+
+        python_logger_name = 'mylogger'
+        message = 'hello world'
+        trace = 'the-project/trace/longlogTraceid'
+
+        record = logging.LogRecord(
+            python_logger_name, logging.INFO,
+            None, None, message, trace, None)
```
Comment (tseaver) [inline logging/tests/unit/handlers/transports/test_background_thread.py]: Python's [`logging.LogRecord`](https://docs.python.org/3/library/logging.html#logging.LogRecord) does not take `trace` or `span_id` as arguments.

File: logging/tests/unit/handlers/transports/test_background_thread.py [hunk 5]
```diff
@@ -64,7 +64,49 @@ def test_send(self):
...
+
+        transport.send(record, message, _GLOBAL_RESOURCE, None)
```
Comment (tseaver) [inline logging/tests/unit/handlers/transports/test_background_thread.py]: Pass `trace` as an argument to `send`, and use named arguments for the optionals.  E.g.:

```python
        transport.send(record, message, resource=_GLOBAL_RESOURCE, trace=trace)
```

File: logging/tests/unit/handlers/transports/test_background_thread.py [hunk 6]
```diff
@@ -64,7 +64,49 @@ def test_send(self):
...
+
+        transport, _ = self._make_one(client, name)
+
+        python_logger_name = 'mylogger'
+        message = 'hello world'
+        span_id = 'the-project/trace/longlogTraceid/span/123456789012abbacdac'
+
+        record = logging.LogRecord(
+            python_logger_name, logging.INFO,
+            None, None, message, None, span_id)
```
Comment (tseaver) [inline logging/tests/unit/handlers/transports/test_background_thread.py]: Python's [`logging.LogRecord`](https://docs.python.org/3/library/logging.html#logging.LogRecord) does not take `trace` or `span_id` as arguments.

File: logging/tests/unit/handlers/transports/test_background_thread.py [hunk 7]
```diff
@@ -64,7 +64,49 @@ def test_send(self):
...
+
+        transport.send(record, message, _GLOBAL_RESOURCE, None)
```
Comment (tseaver) [inline logging/tests/unit/handlers/transports/test_background_thread.py]: Pass `span_id` as an argument to `send`, and use named arguments for the optionals.  E.g.:

```python
        transport.send(record, message, resource=_GLOBAL_RESOURCE, span_id=span_id)
```

**Comment by brunoais** [comment]
@tseaver please re-review.

**Comment by tseaver** [comment]
@brunoais I'm pretty sure we have a new feature release scheduled for the first week in October.  We could make another before then (without that pending feature) if @snehashah16 wants us to.

**Comment by brunoais** [comment]
OK! Thank you!

**Comment by tseaver** [comment]
@brunoais Oops, I crossed my wires in that response:  @snehashah16 is responsible for spanner, not logging.  I think @liyanhui1228 is the logging person.

**Comment by tseaver** [comment]
@brunoais I just added #5924 to track the release of a new `google-cloud-logging` version.

**Comment by brunoais** [comment]
Got it. Thanks for both m8!

**Pull request merged**